python/cuml/internals/memory_utils.py

```diff
@@ -14,92 +14,49 @@
 # limitations under the License.
 #
 
-import contextlib
 import functools
 import operator
 import re
-from dataclasses import dataclass
 from functools import wraps
-from enum import Enum, auto
-
-import cuml
-import cupy as cp
-import numpy as np
-import rmm
-from cuml.common.import_utils import check_min_cupy_version
-from numba import cuda as nbcuda
-from cuml.common.logger import debug
-
-try:
-    from cupy.cuda import using_allocator as cupy_using_allocator
-except ImportError:
-    try:
-        from cupy.cuda.memory import using_allocator as cupy_using_allocator
-    except ImportError:
-        pass
-
-
-class MemoryType(Enum):
-    device = auto(),
-    host = auto()
-    managed = auto()
-    mirror = auto()
 
-    @staticmethod
-    def from_str(memory_type):
-        if isinstance(memory_type, str):
-            memory_type = memory_type.lower()
-
-        try:
-            return MemoryType[memory_type]
-        except KeyError:
-            raise ValueError('Parameter memory_type must be one of "device", '
-                             '"host", "managed" or "mirror"')
+from cuml.internals.global_settings import GlobalSettings
+from cuml.internals.device_support import GPU_ENABLED
+from cuml.internals.mem_type import MemoryType
+from cuml.internals.output_type import (
+    INTERNAL_VALID_OUTPUT_TYPES,
+    VALID_OUTPUT_TYPES
+)
+from cuml.internals.safe_imports import (
+    cpu_only_import_from,
+    gpu_only_import_from,
+    UnavailableNullContext
+)
+
+CudfSeries = gpu_only_import_from('cudf', 'Series')
+CudfDataFrame = gpu_only_import_from('cudf', 'DataFrame')
+cupy_using_allocator = gpu_only_import_from(
+    'cupy.cuda', 'using_allocator', alt=UnavailableNullContext
+)
+PandasSeries = cpu_only_import_from('pandas', 'Series')
+PandasDataFrame = cpu_only_import_from('pandas', 'DataFrame')
+rmm_cupy_allocator = gpu_only_import_from('rmm', 'rmm_cupy_allocator')
 
 
 def set_global_memory_type(memory_type):
```

Comment:
It looks like `set_global_memory_type` and `using_memory_type` are already in `global_settings.py`. Also, `set_global_memory_type` differs slightly here.